src/sidebar/directive/annotation.js

```diff
@@ -222,6 +222,22 @@ function AnnotationController(
 
   /**
     * @ngdoc method
+    * @name annotation.AnnotationController#flag
+    * @description Flag the annotation.
+    */
+  vm.flag = function() {
+    var onRejected = function(reason) {
+      flash.error(
+        errorMessage(reason), 'Flagging annotation failed');
+    };
+    annotationMapper.flagAnnotation(vm.annotation).then(function(){
+      analytics.track(analytics.events.ANNOTATION_FLAGGED);
+      vm.isFlagged = true;
```

Comment:
We'll need to move this state out of the annotation soon, since the client will throw away any local state in the annotation card if you switch tabs or scroll it far enough off screen. For this PR it's OK though.